pkg/encoder_test.go

```diff
@@ -239,14 +254,288 @@ func TestEncoder_EncodeParsed(t *testing.T) {
+		0x33, 0x00, 0x00, 0x00,
+		0x67, // alias
+		0x05, 0x00, 0x00, 0x00,
+		0x61, 0x6c, 0x69, 0x61, 0x73,
+		0x68, // direction
+		0x01, 0x00, 0x00, 0x00,
+		0x02,
+		0x69, // seq
+		0x08, 0x00, 0x00, 0x00,
+		0x2a, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x6a, // sub seq
+		0x00, 0x00, 0x00, 0x00,
+		0x6b, // timestamp
+		0x0c, 0x00, 0x00, 0x00,
+		0xb0, 0xb2, 0x47, 0x64, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00,
+		0x0b, // metadata
+		0x12, 0x00, 0x00, 0x00,
+		0x02, // key
+		0x03, 0x00, 0x00, 0x00,
+		0x6b, 0x65, 0x79,
+		0x02, // value
+		0x05, 0x00, 0x00, 0x00,
+		0x76, 0x61, 0x6c, 0x75, 0x65,
+		0x0c, // protocol
+		0x08, 0x00, 0x00, 0x00,
+		0x70, 0x72, 0x6f, 0x74, 0x6f, 0x63, 0x6f, 0x6c,
+		0x15, // body
+		0x05, 0x00, 0x00, 0x00,
+		0x29, 0x2a, 0x2b, 0x2c, 0x2d,
+
+		0x1e, // parsed msg
+		0x7d, 0x00, 0x00, 0x00,
+		0x0a, // msg id
+		0x45, 0x00, 0x00, 0x00,
+		0x67, // alias
+		0x05, 0x00, 0x00, 0x00,
+		0x61, 0x6c, 0x69, 0x61, 0x73,
+		0x68, // direction
+		0x01, 0x00, 0x00, 0x00,
+		0x02,
+		0x69, // seq
+		0x08, 0x00, 0x00, 0x00,
+		0x2a, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x6a, // sub seq
+		0x12, 0x00, 0x00, 0x00,
+		0x03, // sub seq 1
+		0x04, 0x00, 0x00, 0x00,
+		0x01, 0x00, 0x00, 0x00,
+		0x03, // sub seq 2
+		0x04, 0x00, 0x00, 0x00,
+		0x02, 0x00, 0x00, 0x00,
+		0x6b, // timestamp
+		0x0c, 0x00, 0x00, 0x00,
+		0xb0, 0xb2, 0x47, 0x64, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00,
+		0x0b, // metadata
+		0x12, 0x00, 0x00, 0x00,
+		0x02, // key
+		0x03, 0x00, 0x00, 0x00,
+		0x6b, 0x65, 0x79,
+		0x02, // value
+		0x05, 0x00, 0x00, 0x00,
+		0x76, 0x61, 0x6c, 0x75, 0x65,
+		0x0c, // protocol
+		0x08, 0x00, 0x00, 0x00,
+		0x70, 0x72, 0x6f, 0x74, 0x6f, 0x63, 0x6f, 0x6c,
+		0x0d, // message type
+		0x00, 0x00, 0x00, 0x00,
+		0x1f, // body
+		0x05, 0x00, 0x00, 0x00,
+		0x29, 0x2a, 0x2b, 0x2c, 0x2d,
+
+		0x65, // book
+		0x04, 0x00, 0x00, 0x00,
+		0x62, 0x6f, 0x6f, 0x6b,
+		0x66, // group
+		0x05, 0x00, 0x00, 0x00,
+		0x67, 0x72, 0x6f, 0x75, 0x70,
+	}
+
+	assert.Equal(t, expected, res)
+	assert.Equal(t, len(expected), size)
+}
+
+func TestEncoder_EncodeTwoMsgIntoSeparateGroups(t *testing.T) {
+	e := transport.NewEncoder(nil)
+	rawMsg := transport.RawMessage{
+		MessageId: transport.MessageId{
+			SessionAlias: "alias",
+			Sequence:     42,
+			Direction:    transport.OutgoingDirection,
+			Timestamp:    transport.TimestampFromTime(time.UnixMilli(1682420400000)),
+		},
+		Protocol: "protocol",
+		Metadata: map[string]string{
+			"key": "value",
+		},
+		Body: []byte{41, 42, 43, 44, 45},
+	}
+	parsedMsg := transport.ParsedMessage{
+		MessageId: transport.MessageId{
+			SessionAlias: "alias",
+			Sequence:     42,
+			Direction:    transport.OutgoingDirection,
+			Timestamp:    transport.TimestampFromTime(time.UnixMilli(1682420400000)),
+			Subsequence:  []int32{1, 2},
+		},
+		Protocol: "protocol",
+		Metadata: map[string]string{
+			"key": "value",
+		},
+		CborBody: []byte{41, 42, 43, 44, 45},
```

Comment:
note: Looks like we need to correct the field name in `ParsedMessage` struct. Probably, it should be just `Body` and it should be done in a separate PR

Reply:
Probably, yes